src/features/user/BulkCodes/components/UploadWidget.tsx

```diff
@@ -0,0 +1,173 @@
+import { ReactElement, useCallback, useEffect, useState } from 'react';
+import { useDropzone, ErrorCode } from 'react-dropzone';
+import i18next from '../../../../../i18n';
+
+import FileUploadIcon from '../../../../../public/assets/images/icons/FileUploadIcon';
+import FileProcessingIcon from '../../../../../public/assets/images/icons/FileProcessingIcon';
+import FileAttachedIcon from '../../../../../public/assets/images/icons/FileAttachedIcon';
+
+import { FileImportError, UploadStates } from '../BulkCodesTypes';
+
+import styles from '../BulkCodes.module.scss';
+
+const { useTranslation } = i18next;
+
+interface UploadWidgetInterface {
+  status: UploadStates;
+  onFileUploaded: (fileContents: string) => void;
+  onStatusChange: (newStatus: UploadStates) => void;
+  parseError: FileImportError | null;
+  hasIgnoredColumns: boolean;
+}
+
+const UploadWidget = ({
+  status = 'empty',
+  onStatusChange,
+  onFileUploaded,
+  parseError,
+  hasIgnoredColumns,
+}: UploadWidgetInterface): ReactElement | null => {
+  const { t, ready } = useTranslation(['bulkCodes']);
+  const [error, setError] = useState<FileImportError | null>(null);
+
+  const onDropAccepted = useCallback((acceptedFiles: File[]) => {
+    onStatusChange('processing');
+    const reader = new FileReader();
+    reader.readAsText(acceptedFiles[0]);
+    reader.onabort = () => handleError('file reading was aborted');
+    reader.onerror = () => handleError('file reading has failed');
+    reader.onload = (event: ProgressEvent<FileReader>) => {
+      const csv = event.target?.result;
+      onFileUploaded(csv as string);
+    };
+    setError(null);
+  }, []);
+
+  const onDropRejected = useCallback((fileRejections) => {
+    const error = fileRejections[0].errors[0].code;
+    handleError(error);
+  }, []);
+
+  useEffect(() => {
+    if (parseError) {
+      handleError('parseError', parseError);
+    }
+  }, [parseError]);
+
+  const handleError = (errorType: string, error?: FileImportError) => {
```

Comment:
Try to use different function variable name as we have more than 1 handleError functions in an entire application. Be specific while naming a function.

Reply:
Have renamed this to `handleUploadError`